underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -190,14 +194,22 @@ public boolean delete(String path, boolean recursive) throws IOException {
     final String strippedPath = stripContainerPrefixIfPresent(path);
     Container container = mAccount.getContainer(mContainerName);
     if (recursive) {
+      boolean deletedSelf = false;
+
       // For a file, recursive delete will not find any children
       PaginationMap paginationMap = container.getPaginationMap(
           addFolderSuffixIfNotPresent(strippedPath), DIR_PAGE_SIZE);
       for (int page = 0; page < paginationMap.getNumberOfPages(); page++) {
         for (StoredObject childObject : container.list(paginationMap, page)) {
           deleteObject(childObject);
+          if (childObject.getName().equals(addFolderSuffixIfNotPresent(strippedPath))) {
+            deletedSelf = true;
```

Comment:
This is required because folder suffix is equal to path separator? Should the `paginationMap` call above be `addPathSeparator` and not `folderSuffix` (despite the fact they are equivalent for Swift)?